scripts/engineering/aggregate_phase43_source_root_feasibility.py

```
@@ -0,0 +1,208 @@
+
+def main() -> None:
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--inputs", nargs="+", required=True, type=Path)
+    parser.add_argument("--output-json", required=True, type=Path)
+    parser.add_argument("--output-tsv", required=True, type=Path)
+    args = parser.parse_args()
+
+    if output_paths_conflict(args.output_json, args.output_tsv):
+        parser.error("--output-json and --output-tsv must be distinct paths")
+    if len(args.inputs) < 3:
+        raise SystemExit("expected at least 3 repeated benchmark runs for aggregation")
```

Ruff (0.15.11)

[warning] Avoid specifying long messages outside the exception class

(TRY003)